src/autopas/AutoPas.h

```diff
@@ -101,17 +102,27 @@ class AutoPas {
    *
    */
   void init() {
+    if (_autopasMPICommunicator == AUTOPAS_MPI_COMM_NULL) {
+      AutoPas_MPI_Comm_dup(AUTOPAS_MPI_COMM_WORLD, &_autopasMPICommunicator);
+    }
     _autoTuner = std::make_unique<autopas::AutoTuner<Particle, ParticleCell>>(
         _boxMin, _boxMax, _cutoff, _verletSkin, _verletClusterSize,
         std::move(TuningStrategyFactory::generateTuningStrategy(
             _tuningStrategyOption, _allowedContainers, *_allowedCellSizeFactors, _allowedTraversals,
             _allowedDataLayouts, _allowedNewton3Options, _maxEvidence, _relativeOptimumRange,
-            _maxTuningPhasesWithoutTest, _acquisitionFunctionOption)),
+            _maxTuningPhasesWithoutTest, _acquisitionFunctionOption, _autopasMPICommunicator)),
         _selectorStrategy, _tuningInterval, _numSamples);
     _logicHandler =
         std::make_unique<autopas::LogicHandler<Particle, ParticleCell>>(*(_autoTuner.get()), _verletRebuildFrequency);
   }
 
+  /**
+   * Free the AutoPas MPI communicator.
+   * DO call if: No MPI_Comm was not externally provided or is not freed externally
+   * Do NOT call if: Externally provided MPI_Comm is freed externally
```

Comment:
this is NOT a good idea!
this should always be called consistently.
as you are doing an `AutoPas_MPI_Comm_dup` it should be safe to free it, right?

Reply:
I'm only calling AutoPas_MPI_Comm_dup if has not been set until AutoPas::init() is called.
In finalize I don't know if the communicator has been created by the AutoPas_MPI_Comm_dup in init() or if it has been provided by the client. Thus if I free it blindly, the client implementation would crash if it also attempted to free it.
So either I have these rules or I explicitely store whether the communicator came externally.